fuzzers/aflplusplus/fuzzer.py

```diff
@@ -44,25 +71,91 @@ def build():
     # from writing AFL specific messages to stderr.
     os.environ['AFL_QUIET'] = '1'
 
-    utils.build_benchmark()
+    src = os.getenv('SRC')
+    work = os.getenv('WORK')
+    with restore_directory(src), restore_directory(work):
+        # Restore SRC to its initial state so we can build again without any
+        # trouble. For some OSS-Fuzz projects, build_benchmark cannot be run
+        # twice in the same directory without this.
+        utils.build_benchmark()
+
+    if "cmplog" in build_modes and "qemu" not in build_modes:
+
+        # CmpLog requires an build with different instrumentation.
+        new_env = os.environ.copy()
+        new_env["AFL_LLVM_CMPLOG"] = "1"
+
+        # For CmpLog build, set the OUT and FUZZ_TARGET environment
+        # variable to point to the new CmpLog build directory.
+        build_directory = os.environ['OUT']
+        cmplog_build_directory = get_cmplog_build_directory(build_directory)
+        os.mkdir(cmplog_build_directory)
+        new_env['OUT'] = cmplog_build_directory
+        fuzz_target = os.getenv('FUZZ_TARGET')
+        if fuzz_target:
+            new_env['FUZZ_TARGET'] = os.path.join(cmplog_build_directory,
+                                                  os.path.basename(fuzz_target))
+
+        print('Re-building benchmark for CmpLog fuzzing target')
+        utils.build_benchmark(env=new_env)
+
     shutil.copy('/afl/afl-fuzz', os.environ['OUT'])
 
 
 def fuzz(input_corpus, output_corpus, target_binary):
     """Run fuzzer."""
+    # Calculate CmpLog binary path from the instrumented target binary.
+    target_binary_directory = os.path.dirname(target_binary)
+    cmplog_target_binary_directory = (
+        get_cmplog_build_directory(target_binary_directory))
+    target_binary_name = os.path.basename(target_binary)
+    cmplog_target_binary = os.path.join(cmplog_target_binary_directory,
+                                        target_binary_name)
+
     afl_fuzzer.prepare_fuzz_environment(input_corpus)
+    os.environ['AFL_PRELOAD'] = '/afl/libdislocator.so'
+
+    flags = ["-d"]  # FidgetyAFL is better when runnign alone
```

Comment:
nit: typo in "running".
Also I'm not sure what the comment means. Does it mean that when starting from no seed fidgety AFL is better (I think this is the case).

Reply:
I mean when running alone without other parallel instances of AFL++.

Reply:
Ah got it!